From 0997c68b00be46c267b5fbb92d8f5ff404b780a3 Mon Sep 17 00:00:00 2001
From: Shuhao Wu <shuhao@shuhaowu.com>
Date: Thu, 13 Dec 2018 14:33:03 -0500
Subject: [PATCH] Allow RuneliteAPI url be configurable

So it can be locally overwritten by a launcher via an environment
variable.
---
 .../java/net/runelite/http/api/RuneLiteAPI.java    | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/http-api/src/main/java/net/runelite/http/api/RuneLiteAPI.java b/http-api/src/main/java/net/runelite/http/api/RuneLiteAPI.java
index bc6721518..b73758d20 100644
--- a/http-api/src/main/java/net/runelite/http/api/RuneLiteAPI.java
+++ b/http-api/src/main/java/net/runelite/http/api/RuneLiteAPI.java
@@ -71,7 +71,12 @@ public class RuneLiteAPI
 
 	public static HttpUrl getApiBase()
 	{
-		return HttpUrl.parse(BASE + getVersion());
+		String base = System.getenv("RUNELITE_API_BASE");
+		if (base == null)
+		{
+			base = BASE;
+		}
+		return HttpUrl.parse(base + getVersion());
 	}
 
 	public static HttpUrl getStaticBase()
@@ -81,7 +86,12 @@ public class RuneLiteAPI
 
 	public static String getWsEndpoint()
 	{
-		return WSBASE + getVersion() + "/ws";
+		String wsbase = System.getenv("RUNELITE_WS_BASE");
+		if (wsbase == null)
+		{
+			wsbase = WSBASE;
+		}
+		return wsbase + getVersion() + "/ws";
 	}
 
 	public static String getVersion()
-- 
2.19.2

